torch/testing/_comparison.py

```diff
@@ -333,25 +333,36 @@ def __init__(
         self.id = id
         self._unknown_parameters = unknown_parameters
 
+    @staticmethod
+    def _inputs_not_supported() -> NoReturn:
+        raise UnsupportedInputs()
+
     @staticmethod
     def _check_inputs_isinstance(*inputs: Any, cls: Union[Type, Tuple[Type, ...]]):
         """Checks if all inputs are instances of a given class and raise :class:`UnsupportedInputs` otherwise."""
         if not all(isinstance(input, cls) for input in inputs):
-            raise UnsupportedInputs()
+            Pair._inputs_not_supported()
 
-    def _make_error_meta(self, type: Type[Exception], msg: str) -> ErrorMeta:
-        """Makes an :class:`ErrorMeta` from a given exception type and message and the stored id.
+    def _fail(
```

Comment:
Two changes here:

1. For the same reason as above, this leaked details.
2. Just creating the exception but leaving the `raise` to the caller is fine in theory but one can easily botch this. For example, I myself forgot to `raise` it in a custom comparison in `torchvision`. It took us quite a while to even realize that we have unintentionally disabled part of the comparison and in turn we needed to untangle the mess from adding tests that should have failed but didn't. See pytorch/vision#6934.